tools/payloads/ysoserial/findYsoserialOffsets.rb

```diff
@@ -0,0 +1,179 @@
+  stdout, stderr, status = Open3.capture3("java -jar #{program} #{payloadName} '#{searchString}'")
+  
+  payload = stdout
+  payload.force_encoding("binary")
+
+  if payload.length==0 and stderr.length>0
+    # Pipe errors out to the console
+    STDERR.puts stderr.split("\n").each {|i| i.prepend("    ")}
+  elsif stderr.include?"java.lang.IllegalArgumentException"
+    #STDERR.puts "  WARNING: '#{payloadName}' with #{program} requires complex args and may not be supported"
+    return nil
+  elsif stderr.include?"Error while generating or serializing payload"
+    #STDERR.puts "  WARNING: '#{payloadName}' with #{program} errored and may not be supported"
+    return nil
+  elsif stdout == "\xac\xed\x00\x05\x70"
+    #STDERR.puts "  WARNING: '#{payloadName}' with #{program} returned null and may not be supported"
+    return nil
+  else
+    #STDERR.puts "  Successfully generated #{payloadName} using #{program}"
+
+    # Strip out the semi-randomized ysoserial string and trailing newline
+    payload.gsub!(/#{YSOSERIAL_RANDOMIZED_HEADER}[[:digit:]]+/, 'ysoserial/Pwner00000000000000')
+    return payload
+  end
+end
+
+def generatePayloadArray(payloadName)
+  # Generate and return a number of payloads, each with increasingly longer strings, for future comparison
+  payloadArray = []
+  (1..PAYLOAD_TEST_MAX_LENGTH).each do |i|
+    payload = generatePayload(payloadName,i)
+    return nil if payload.nil?
+    payloadArray[i] = payload
+  end
+  return payloadArray
+end
+
+def isLengthOffset?(currByte,nextByte)
+  # If this byte has been changed, and is different by one, then it must be a length value
+  if nextByte and currByte.position == nextByte.position and currByte.action == "-"
+    if nextByte.element.ord - currByte.element.ord == 1
+      #STDERR.puts "Found a length offset at #{currByte.position}"
+      return true
+    end
+  end
+  return false
+end
+
+def isBufferOffset?(currByte,nextByte)
+  # If this byte has been inserted, then it must be part of the increasingly large payload buffer
+  if (currByte.action == "+" and (nextByte.nil? or (currByte.position != nextByte.position)))
+    #STDERR.puts "Found a buffer offset at #{currByte.position}"
+    return true
+  end
+  return false
+end
+
+def diff(a,b)
+  diffs = []
+  obj=  Diff::LCS.diff(a,b)
+  obj.each do |i|
+    i.each do |j|
+      diffs.push(j)
+    end
+  end
+  return diffs
+end
+
+def getPayloadList()
+  # Call ysoserial and return the list of payloads that can be generated
+  payloads = `java -jar ysoserial-original.jar 2>&1`
+  payloads.encode!('ASCII', 'binary', invalid: :replace, undef: :replace, replace: '')
+  payloads = payloads.split("\n")
+
+  # Make sure the headers are intact, then skip over them
+  abort unless payloads[0] == "Y SO SERIAL?"
+  payloads = payloads.drop(5)
+
+  payloadList = []
+  # Skip the header rows
+  payloads.each do |line|
+    next unless line.include?"     "
```

Comment:
I'm guessing, based on the next line, that you desire `line` to `start_with? '     '`, rather than `include? '     '`

```suggestion
    next unless line.start_with?"     "
```